### PR TITLE
Improve spring readme

### PR DESCRIPTION
**Pull request by gurbuzali**

### CODE REVIEW

File: spring/README.md [hunk 1]
```diff
@@ -3,27 +3,80 @@
...
+
+## Spring Boot Integration
+
+You can integrate Hazelcast Jet with Spring Boot easily.
+The approach is very similar to the programmatic configuration.
+You need to have a 
+[config class](src/main/java/jet/spring/config/AppConfig.java)
+which has `@SpringBootApplication` annotation. 
+
+```java
```
Comment (vladoschreiner) [inline spring/README.md]: The code sample however doesn't contain @SpringBootApplication annotation, so it's a little bit confusing:)

File: spring/README.md [hunk 2]
```diff
@@ -3,27 +3,80 @@
...
+}
+```
 
 ## @SpringAware annotation
 
-You can enable `@SpringAware` annotation to auto-wire beans to processors. 
+You can enable `@SpringAware` annotation to auto-wire beans to
+[processors](src/main/java/jet/spring/SpringBootSample.java). 
 You have to configure Hazelcast Jet to use `SpringManagedContext`.
+
```
Comment (vladoschreiner) [inline spring/README.md]: Could you please add a short information about when is this useful? E.g. "you define Jet Job on your client, it get's executed in a remote cluster though. One can use @SpringAware to get a remote Spring Context injected to a running Jet Job".

File: spring/README.md (hunk 1 above)
Comment (gurbuzali) [reply spring/README.md]: it is over here https://github.com/hazelcast/hazelcast-jet-code-samples/blob/master/spring/src/main/java/jet/spring/config/AppConfig.java#L39

File: spring/README.md (hunk 2 above)
Comment (gurbuzali) [reply spring/README.md]: Even if you have a single node cluster and you are submitting the jobs from that member instance you need to use `@SpringAware` annotation to inject spring beans to your processors. So this is not a remote cluster specific issue. Jet creates processors (via suppliers) itself so you need a way to initialize them with spring-context and `@SpringAware` annotation gives you this ability

**Pull request merged**